screens/ProductionScreens/BudgetAddExpense.jsx

```diff
@@ -0,0 +1,181 @@
+        const url = "";
+        if (receiptURI) {
+            try {
+                const blob = receiptURI.blob();
+                const extension = receiptURI.substring(receiptURI.lastIndexOf("."));
+                const snapshot = await uploadBytes(
+                    ref(storageRef, "receipts/" + expenseID + extension),
+                    blob
+                );
+                url = await getDownloadURL(snapshot.ref);
+            } catch (error) {
+                setAlert(
+                    "Error uploading receipt",
+                    "bg-red-500",
+                    icon({ name: "circle-exclamation" })
+                );
+            }
+        }
+    };
+
+    const setReceiptLibrary = async () => {
+        const libraryStatus = await requestMediaLibraryPermissionsAsync();
+        if (libraryStatus.status != "granted") {
+            setAlert(
+                "Please grant permission to access your photo library to upload a receipt",
+                "bg-red-500",
+                icon({ name: "circle-exclamation" })
+            );
+            return;
+        }
+        launchImageLibraryAsync({ quality: 0.1 }).then((response) => {
+            if (!response.canceled) {
+                try {
+                    fetch(response.assets[0].uri).then(() => {
+                        setReceiptURI(response.assets[0].uri);
+                    });
+                } catch (error) {
+                    setAlert(
+                        "An error occurred while uploading the receipt. Please try again.",
+                        "bg-red-500",
+                        icon({ name: "circle-exclamation" })
+                    );
+                }
+            }
+        });
+    };
+
+    const setReceiptCamera = async () => {
+        const cameraStatus = await requestCameraPermissionsAsync();
+        if (cameraStatus.status != "granted") {
+            setAlert(
+                "Please grant permission to access your camera to upload a receipt",
+                "bg-red-500",
+                icon({ name: "circle-exclamation" })
+            );
+            return;
+        }
+        launchCameraAsync({ quality: 0.1 }).then((response) => {
+            if (!response.canceled) {
+                try {
+                    fetch(response.assets[0].uri).then(() => {
+                        setReceiptURI(response.assets[0].uri);
+                    });
+                } catch (error) {
+                    setAlert(
+                        "An error occurred while uploading the receipt. Please try again.",
+                        "bg-red-500",
+                        icon({ name: "circle-exclamation" })
+                    );
+                }
+            }
+        });
+    };
+
+    return (
+        <View className="py-2 flex justify-center items-center">
+            <Text className="self-center text-3xl font-extrabold">Add Expense</Text>
+            <ScrollView className="h-5/6 w-10/12">
```

Comment:
Wish there was a tap to dismiss keyboard here like the modals

Reply:
Fixed I believe. You can't click on the title, but everything else is fine, let me know if you think that is okay